Source/Core/Core/PowerPC/GDBStub.cpp

```diff
@@ -31,6 +32,29 @@ typedef SSIZE_T ssize_t;
 #include "Core/PowerPC/PPCCache.h"
 #include "Core/PowerPC/PowerPC.h"
 
+namespace
+{
+#ifdef _WIN32
+class WSA_RAII
```

Comment:
I'd put this in a separate header in Common/ (SocketContext?) and put the ifdefs around the function calls and the member, like this:

```cpp
namespace Common
{
class SocketContext
{
public:
#ifdef _WIN32
  SocketContext() { static_cast<void>(...); }
  ~SocketContext() { ... }
#else
  SocketContext() = default;
#endif
  // ...

private:
#ifdef _WIN32
  WSADATA m_data;
#endif
};
}  // namespace Common
```

The advantage of doing it like this is that it would abstract the WSA thing away from everything that just wants to use sockets. You would be able to remove all the ifdefs from GDBStub.cpp but also from IOS/Network/IP/Top.cpp.

Reply:
Nice.  Yeah, I thought of this too but wasn't sure what was more preferred (always having the object or only on windows).  I can definitely make this change!